registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspCsvServiceImpl.java

```diff
@@ -64,8 +67,68 @@ public AspspCsvServiceImpl(AspspRepository aspspRepository, AspspCsvRecordConver
     @Override
     public void importCsv(byte[] file) {
         List<AspspPO> aspsps = readAllRecords(file);
-        aspspRepository.deleteAll();
-        aspspRepository.saveAll(aspsps);
+        aspspRepository.delete();
+        aspspRepository.saveAll(uuidGeneratorService.checkAndUpdateUUID(aspsps));
+    }
+
+    @Override
+    public void deserializeAndMerge(byte[] file) {
+        List<AspspPO> input = readAllRecords(file);
+        List<AspspPO> database = aspspRepository.findAll();
+        List<AspspPO> forDeleting = new LinkedList<>(), forSave = new LinkedList<>();
+
+        database.forEach(item -> {
+            input.forEach(element -> {
+                if (element.getId() == null) {
+                    logicForProcessingWithNullId(input, forSave, item, element);
+//                no match by id, but match by BIC and BLZ
+                } else if (areBicAndBlzEqualWithDifferentId(item, element)) {
+                    forDeleting.add(item);
+                }
+
+//                no match, match by id, or id is NULL and no match by BIC and BLZ
+                forSave.add(element);
```

Comment:
what about the case when id are equals?

Reply:
Those records will be added into forSave list and saveAll(), that's below, will handle with them. 

Should it be some separate logic?

Reply:
got it